workflow/automation/metadata/collect_metadata.py

```diff
@@ -0,0 +1,287 @@
+                states.extend(
+                    db.execute(
+                        "SELECT * from state WHERE run_name=? AND status != ? AND proc_type=? AND last_modified>?",
+                        (
+                            rel_name,
+                            const.Status.created.value,
+                            proc_type,
+                            failed_task_modified_time,
+                        ),
+                    ).fetchall()
+                )
+            else:
+                # There were no failed tasks for this proc_type
+                states.extend(
+                    db.execute(
+                        "SELECT * from state WHERE run_name=? AND status != ? AND proc_type=?",
+                        (rel_name, const.Status.created.value, proc_type),
+                    ).fetchall()
+                )
+    else:
+        states = db.execute(
+            "SELECT * from state WHERE run_name=? AND status != ?",
+            (rel_name, const.Status.created.value),
+        ).fetchall()
+
+    # DB Metadata
+    for state in states:
+        # Get proc_type and job_id
+        _, _, proc_type, status, job_id, _ = state
+        proc_type_name = db.execute(
+            "SELECT proc_type from proc_type_enum where id=?",
+            (proc_type,),
+        ).fetchone()[0]
+        proc_type_name = (
+            "IM_calc" if proc_type_name == "IM_calculation" else proc_type_name
+        )
+        if proc_type_name in METADATA_PROC_TYPES:
+            (
+                _,
+                _,
+                queued_time,
+                start_time,
+                end_time,
+                nodes,
+                cores,
+                memory,
+                WCT,
+            ) = db.execute(
+                "SELECT * from job_duration_log WHERE job_id=?",
+                (job_id,),
+            ).fetchone()
+            # Add runtime and cores to df
+            runtime = end_time - start_time
+            df.loc[rel_name, f"{proc_type_name}_runtime"] += runtime / 60
+            df = add_db_stat(df, rel_name, f"{proc_type_name}_cores", cores)
+            # Add to core hours
+            df.loc[rel_name, f"{proc_type_name}_core_hours"] += cores * runtime / 3600
+            df.loc[rel_name, f"Total_core_hours"] += cores * runtime / 3600
+            # Add to resubmits dict
+            if resub_counter.get(proc_type_name) is None:
+                resub_counter[proc_type_name] = 0
+            else:
+                resub_counter[proc_type_name] += 1
+    # Add resubmits to df
+    for k, v in resub_counter.items():
+        df.loc[rel_name, f"{k}_resubmits"] = v
+
+    # Extra Metadata
+    df.loc[rel_name, "IM_calc_components"] = len(params["ims"]["component"])
+    df.loc[rel_name, "EMOD3D_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "HF_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "BB_nt"] = params["sim_duration"] / params["dt"]
```

Comment:
![image](https://user-images.githubusercontent.com/25143301/171530349-899d5819-5563-47c6-9891-75d281a0baf1.png)


Just to be confusing there are 3 dts - HF / BB as in the subsection and looks like emod has the global one.

Reply:
IM_calc uses BB DT